src/IcalParser.php

```diff
@@ -573,23 +574,25 @@ public function getEvents($includeRecurring=false) {
 
                             if(!empty($this->data["_RECURRENCE_COUNTERS_BY_UID"][$modifiedEventUID])) {
                                 $counter = $this->data[ "_RECURRENCE_COUNTERS_BY_UID" ][ $modifiedEventUID ];
-                                $originalEvent = $this->data[ "VEVENT" ][ $counter ];
-                                if(!empty($originalEvent[ 'RECURRENCES' ]) && isset($originalEvent[ 'SEQUENCE' ])) {
-                                    $originalEventSeq = intval($originalEvent['SEQUENCE'], 10);
-                                    if($modifiedEventSeq >= $originalEventSeq) {
-                                        for ($j = 0; $j < count($originalEvent['RECURRENCES']); $j++) {
-                                            $recurDate = $originalEvent[ 'RECURRENCES' ][$j];
-                                            $formatedStartDate = $recurDate->format('Ymd\THis');
-                                            if($formatedStartDate === $modifiedEventRecurID) {
-                                                unset($this->data[ "VEVENT" ][ $counter ]['RECURRENCES'][$j]);
-                                                $this->data["VEVENT"][$counter]['RECURRENCES'] = array_values($this->data["VEVENT"][$counter]['RECURRENCES']);
-                                                break;
+                                if(isset($counter) && (in_array($counter, $this->data["VEVENT"]))) {
+                                    $originalEvent = $this->data[ "VEVENT" ][ $counter ];
```

Comment:
Because I am filtering out events older than 1yr ago or newer than 1yr in the future, there are times when a "recurrence" of an event is in the array of events, but the original event is not. Just nested this logic in an if statement to handle those cases when the original event doesn't exist in the filtered list of events.